eventually/authentication/views.py

```diff
@@ -5,16 +5,109 @@
 
 from django.views.generic.base import View
 from django.contrib.auth import authenticate, login, logout
-from django.http import HttpResponse
+from django.db import IntegrityError
+from django.http import HttpResponse, JsonResponse
 from authentication.models import CustomUser
 from utils.utils import json_loads
-from utils.passwordreseting import send_reseting_letter, reset_pasword
+from utils.passwordreseting import send_reseting_letter, reset_password
 from utils.validators import password_validator, email_validator, data_validator
 from utils.jwttoken import handle_token
 
 
 class UserView(View):
-    """UserView view handles GET, POST, PUT, DELETE requests."""
+    """    A class to handle GET, PUT and DELETE operations    """
+
+    def get(self, request, user_id):
+        """
+        Handles GET request, retrieves a user profile from the database
+        :param request: request from the web page
+        :type request: WSGIRequest
+
+        :param user_id: id of a user to return.
+        :type user_id: int
+
+        :return: user dictionary
+            E.G.
+                {
+                "id": 24,
+                "first_name": "TechGG76",
+                "middle_name": "Postgres",
+                "last_name": "Jr.",
+                "email": "postman424@co.uk",
+                "created_at": 1509831239,
+                "updated_at": 1509835437
+                }
+        """
+
+        user = CustomUser.get_by_id(user_id=user_id)
+        if user:
+            user = user.to_dict()
+            return JsonResponse(user, status=200)
+        return HttpResponse('User does not exist', status=404)
+
+
+    def put(self, request, user_id):
+        """
+        Handles PUT request, modifies the user profile in the datatbase.
+
+        :param request: request from the web page with a json containing changes to be applied
+        :type request: WSGIRequest
+        :Example: incoming JSON request:
+        | {
+        |   "email":"pss@com.ua",
+        |   "last_name":"Ariana"
+        | }
+
+        :param user_id: id of a user which has to be changed
+        :type user_id: int
+
+        :return: status 200 if user has been updated,
+                 status 400 if user hasn't been updated,
+        """
+
+        new_attrs = json_loads(request.body)
+        if new_attrs:
+            user_to_modify = CustomUser.get_by_id(user_id)
```

Comment:
add verification is request.user is user_to_modify.

Reply:
Fixed, my bad.